group-coordinator/src/main/java/org/apache/kafka/coordinator/group/modern/ModernGroup.java

```diff
@@ -398,6 +423,21 @@ public Map<String, TopicMetadata> computeSubscriptionMetadata(
         return Collections.unmodifiableMap(newSubscriptionMetadata);
     }
 
+    public static long computeMetadataHash(
+        Map<String, SubscriptionCount> subscribedTopicNames,
+        Map<String, Long> topicHashCache,
+        MetadataImage metadataImage
+    ) {
+        Map<String, Long> topicHash = new HashMap<>(subscribedTopicNames.size());
+        subscribedTopicNames.keySet().forEach(topicName ->
+            topicHash.put(
+                topicName,
+                topicHashCache.computeIfAbsent(topicName, k -> Utils.computeTopicHash(topicName, metadataImage))
```

Comment:
We may have an issue here. If the topic does not exist, we cache zero. However, we don't remove it when the topic is created, isn't it? Or, is it covered by `topicsDelta.changedTopics()`? It would be great if we could add more tests.

We should also add unit tests for this new method.